cfme/common/provider.py

```diff
@@ -146,6 +163,10 @@ def mgmt(self):
         return self.get_mgmt_system()
 
     @property
+    def collection(self):
+        return ProviderList(type(self), self.appliance)
```

Comment:
I guess this could be DRY-ed out so you just specify the List class or something :)